src/taskEvents/script.js

```diff
@@ -0,0 +1,283 @@
+    innerText: 'Choose a skill to add',
+  });
+  const skillCategorySelect = createElement({
+    type: 'select',
+    attributes: { class: 'skill-category-select' },
+  });
+
+  for (let i = 0; i < skills.length; i++) {
+    const option = createElement({
+      type: 'option',
+      attributes: { class: 'options' },
+      innerText: skills[i],
+    });
+    skillCategorySelect.appendChild(option);
+  }
+
+  skillCategoryDiv.appendChild(skillCategoryLabel);
+  skillCategoryDiv.appendChild(skillCategorySelect);
+
+  const skillLevelDiv = createElement({
+    type: 'div',
+    attributes: { class: 'skill-level-div' },
+  });
+  const skillLevelLabel = createElement({
+    type: 'p',
+    attributes: { class: 'skill-level-label' },
+    innerText: 'choose skill level',
+  });
+  const skillLevelSelect = createElement({
+    type: 'select',
+    attributes: { class: 'skill-level-select' },
+  });
+
+  for (let i = 0; i < level.length; i++) {
+    const option = createElement({
+      type: 'option',
+      attributes: { class: 'option' },
+      innerText: level[i],
+    });
+    skillLevelSelect.appendChild(option);
+  }
+
+  skillLevelDiv.appendChild(skillLevelLabel);
+  skillLevelDiv.appendChild(skillLevelSelect);
+
+  const submitBtn = createElement({
+    type: 'button',
+    attributes: { class: 'submit-btn' },
+    innerText: 'Add Skill',
+  });
+
+  // once we have the tags collection we can make an API call here to add that skill tag to user, when the super user clicks on submit
+
+  containerDiv.appendChild(mainTitle);
+  containerDiv.appendChild(skillCategoryDiv);
+  containerDiv.appendChild(skillLevelDiv);
+  containerDiv.appendChild(submitBtn);
+}
+
+async function openUserActivityModal(username) {
+  document.getElementById('close-btn2').disabled = true;
+  document.getElementById('generic-modal').style.visibility = 'visible';
+  const containerDiv = document.getElementById('container-div');
+
+  await renderCard(containerDiv, `${username}'s Task Logs`, username);
+  document.getElementById('close-btn2').disabled = false;
+}
+
+async function renderCard(container, title, username, isAllTasks) {
+  try {
+    addLoader(container);
+    const mainTitle = createElement({
+      type: 'h1',
+      attributes: { class: 'main-title' },
+      innerText: title,
+    });
+    const { logs } = await getTaskLogs(username);
+    const promises = logs.map((log) => getData(log));
+    const allTaskData = await Promise.all(promises);
+    allTaskData.map((data) =>
+      createEventCard(
+        container,
+        data.title,
+        data.message,
+        data.purpose,
+        data.userName,
+        data.taskLevel.category,
+        data.taskLevel.level,
+        isAllTasks,
+        createProfileModal,
+      ),
+    );
+    container.prepend(mainTitle);
+  } catch (error) {
+    // in case our API call fails we need to enable that button
+    document.getElementById('closeBtn2').disabled = false;
```

Comment:
Rather than **closeBtn2** It would be good to have some other name here

Reply:
will do that